apis/apps/v1alpha1/resourcedistribution_types.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2021 The Kruise Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ResourceDistributionSpec defines the desired state of ResourceDistribution.
+type ResourceDistributionSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Resource must be the complete yaml that users want to distribute
+	Resource runtime.RawExtension `json:"resource,omitempty"`
+
+	// WritePolicy define the behavior when other resources with the same name already exist in target namespaces
+	WritePolicy ResourceDistributionWritePolicyType `json:"writePolicy,omitempty"`
+
+	// Targets defines the namespaces that users want to distribute to
+	Targets ResourceDistributionTargets `json:"targets,omitempty"`
+}
+
+// ResourceDistributionWritePolicyType defines the behaviors,
+// when Resource has the same kind and name with existing resources.
+type ResourceDistributionWritePolicyType string
```

Comment:
1. If there is a resource with the same name, is it simply not allowed to be created?
2. For distributed resources, what if the user edits directly?

Reply:
**If there is a resource with the same name, is it simply not allowed to be created?**
- If `writePolicy` is `strict`:
   - The resource will be distributed **iff** there is no conflict (distribute to all, or none of them);
   - Webhooks will intercept the create request
- If `writePolicy` is `overwrite`:
   - The existing resources with the same name will be overwritten;
   - Webhooks **will not** intercept the create request
- If `writePolicy` is `ignore`:
   - The resource will not be distributed to the conflicting namespaces (but will be distributed to the others);
    - Webhooks **will not** intercept the create request
  
**For distributed resources, what happens if the user edits or deletes directly?**
- ResourceDistribution will do nothing if the user edits or deletes resource directly.
- The resource synchronization can only be done by modifying ResourceDistribution.Spec.Resource.

3 I agree

Reply:
Thanks very much! I adopted your useful suggestion, and remove `writePolicy` to simplify API. 
The behavior of ResourceDistribution will always be `strict` after this simplification.